src/DonorDashboards/Admin/Settings.php

```diff
@@ -151,26 +151,57 @@ public function generateDonorDashboardPage() {
 	 *
 	 * @since 2.10.0
 	 */
-	protected function getDonorDashboardPageContent( $format ) {
-
-		switch ( $format ) {
-			case 'block': {
-				return get_comment_delimited_block_content(
-					'give/donor-dashboard',
-					[
-						'align' => 'wide',
-					],
-					null
-				);
-			}
-			case 'shortcode': {
-				return '[give_donor_dashboard]';
-			}
-			default: {
-				return null;
-			}
+	protected function getDonorDashboardPageContent() {
+
+		if ( $this->isBlockEditorActive() ) {
+			return get_comment_delimited_block_content(
+				'give/donor-dashboard',
+				[
+					'align' => 'wide',
+				],
+				null
+			);
+		} else {
+			return '[give_donor_dashboard]';
+		}
+
+	}
+
+	/**
+	 * Determine whether the block editor is active
+	 *
+	 * @return bool
+	 *
+	 * @since 2.10.0
+	 */
+	private function isBlockEditorActive() {
```

Comment:
It looks like this function is doing more than one thing - could you separate out the logic so that each function does "one thing"? This would also clarify what is specifically being checked.

I see:

- Is Gutenberg Active (thus also installed)
- Is WP version greater than 5.0-beta.
- Is the Classic Editor active (thus also installed)

Reply:
Done! Ref: https://github.com/impress-org/givewp/pull/5739/commits/56f856aa047246146112138ccd378a06041277c9

Reply:
I personally use the Disable Gutenberg plugin on client sites (because the Classic editor plugin puts "upsells" to try gutenberg that I don't want my clients to see in some cases). 

Also, classic editor has lots of conditional logic where the plugin itself can be active, but only restricting use on Pages, not posts, etc. 

What I found was that the front-end still parses the block markup even with Gutenberg disabled.

Reply:
@henryholtgeerts it sounds like this changes our algorithm, so let's consider @Benunc's feedback for a refactor.

Reply:
I've added a check for the Disable Gutenberg plugin as well, so that the Donor Dashboard page is genretated with shortcode if that plugin is enabled.

I do not believe it is worth our time to account for conditional setting inside either the Classic Editor plugin or Disable Gutenberg, since
1. They each use their own unique methods/logic for storing these conditions and our development for this edge case should not be bound up in keeping current with their decision making.
2. Admins who have these plugins installed are likely familiar with shortcodes, and will not be confused to see one appear in the block editor, if they have indeed disabled the block editor for certain post types but not all.

I think adding a check for the Disable Gutenberg plugin is warranted, since itself and Classic Editor are by a wide margin the two most popular methods of disabling the block editor. 

Currently, the algorithm works like this:

If a site does not have Classic Editor activated AND it does not have Disable Gutenberg activated AND ( it is running WP 5.0+ OR it has the Gutenberg plugin installed ), it will load the page via a block. Otherwise, it will load the page via a shortcode.